gateway/field_presence.go

```diff
@@ -0,0 +1,109 @@
+package gateway
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/generator"
+	"google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const fieldPresenceMetaKey = "field-paths"
```

Comment:
should this be some kind of object or something? I think it's best practice to avoid using anything that could collide as a key in the metadata. Or is this avoided somehow?

Reply:
It is a metadata key, not a context key, so it has to be a string (and hope/arrange for no collision). Context keys are arbitrary types, but the `metadata.MD` is a `map[string][]string`